test/modules/TestResultCoordinator/Reports/EdgeHubRestartTest/EdgeHubRestartMessageReportGenerator.cs

```diff
@@ -0,0 +1,286 @@
+        {
+            Logger.LogInformation($"Generating report: {nameof(EdgeHubRestartMessageReport)} for [{this.SenderSource}] and [{this.ReceiverSource}]");
+
+            bool isDiscontinuousSequenceNumber = false;
+            long previousSeqNum = 0;
+            ulong passedMessageCount = 0;
+            ulong senderMessageCount = 0;
+            ulong receiverMessageCount = 0;
+            bool hasSenderResult = false;
+            long senderSeqNum = 0;
+            bool hasReceiverResult = false;
+            long receiverSeqNum = 0;
+
+            (senderMessageCount, hasSenderResult, senderSeqNum) =
+                await this.MoveNextSenderResultAsync(senderMessageCount);
+
+            (receiverMessageCount, hasReceiverResult, receiverSeqNum) =
+                await this.MoveNextReceiverResultAsync(receiverMessageCount);
+
+            while (hasSenderResult && hasReceiverResult)
+            {
+                this.ValidateResult(
+                    this.SenderTestResults.Current,
+                    this.SenderSource,
+                    TestOperationResultType.EdgeHubRestartMessage.ToString());
+
+                this.ValidateResult(
+                    this.ReceiverTestResults.Current,
+                    this.ReceiverSource,
+                    TestOperationResultType.Messages.ToString());
+
+                if (receiverSeqNum > senderSeqNum)
+                {
+                    // Increment sender result to have the same seq as the receiver
+                    (senderMessageCount, hasSenderResult, senderSeqNum) = await this.IterateResultToSequenceNumberAsync(
+                        hasSenderResult,
+                        this.MoveNextSenderResultAsync,
+                        receiverSeqNum,
+                        senderMessageCount);
+                }
+
+                if ((receiverSeqNum < senderSeqNum) && hasSenderResult)
+                {
+                    // Increment receiver result to have the same seq as the sender
+                    (receiverMessageCount, hasReceiverResult, receiverSeqNum) = await this.IterateResultToSequenceNumberAsync(
+                        hasReceiverResult,
+                        this.MoveNextReceiverResultAsync,
+                        senderSeqNum,
+                        receiverMessageCount);
+                }
+
+                if (!hasSenderResult || !hasReceiverResult)
+                {
+                    break;
+                }
+
+                bool isCurrentMessagePassing = this.VerifyCurrentResult(
+                    senderSeqNum,
+                    receiverSeqNum);
+
+                // Make sure the sequence number is incremental
+                isDiscontinuousSequenceNumber |= (previousSeqNum + 1) != senderSeqNum;
+                previousSeqNum++;
+
+                // If this message passed, increment the count for a good message
+                passedMessageCount += isCurrentMessagePassing ? 1UL : 0UL;
```

Comment:
if isCurrentMessagePassing is used only here, please remove it and directly call this.VerifyCurrentResult method.